finat/quadrature_element.py

```diff
@@ -57,6 +59,10 @@ def value_shape(self):
     def fiat_equivalent(self):
         ps = self._rule.point_set
         weights = getattr(self._rule, 'weights', None)
+        if weights is None:
+            # we need the weights.
+            quad_rule = FIAT.create_quadrature(self.cell, self._degree, self._scheme)
+            weights = quad_rule.get_weights()
```

Comment:
This is the wrong bandage, I think. It also means that if we ever change `make_quadrature` then we have to remember a change here too.

I think the right thing to do is to turn the symbolic weights_expression of the `rule` into an array. Can you try this patch against trunk instead?

```diff
diff --git a/finat/quadrature_element.py b/finat/quadrature_element.py
index 33622d6..0ef3343 100644
--- a/finat/quadrature_element.py
+++ b/finat/quadrature_element.py
@@ -6,6 +6,7 @@ import FIAT
 
 import gem
 from gem.utils import cached_property
+from gem.interpreter import evaluate
 
 from finat.finiteelementbase import FiniteElementBase
 from finat.quadrature import make_quadrature
@@ -56,7 +57,8 @@ class QuadratureElement(FiniteElementBase):
     @cached_property
     def fiat_equivalent(self):
         ps = self._rule.point_set
-        weights = getattr(self._rule, 'weights', None)
+        weights, = evaluate([self._rule.weight_expression])
+        weights = weights.arr.flatten()
         return FIAT.QuadratureElement(self.cell, ps.points, weights)
 
     def basis_evaluation(self, order, ps, entity=None, coordinate_mapping=None):
```

Reply:
Excellent, that also works, thank you!

I've updated the PR with the gist of this patch.